go.mod

```diff
@@ -49,6 +49,7 @@ require (
 	golang.org/x/sync v0.6.0
 	google.golang.org/grpc v1.61.0
 	google.golang.org/protobuf v1.32.0
+	resenje.org/singleflight v0.4.1
```

Comment:
**Note:** this third-party package is intentional and necessary because it allows context separation between the resolver's parent context and the singleflight callback's context. This prevents a parent subproblem from cancelling an in-flight child subproblem that needs to resolve for an adjacent branch.